src/raftstore/store/fsm/peer.rs

```diff
@@ -599,31 +628,9 @@ impl<T: Transport, C: PdClient> Store<T, C> {
             return Ok(Some(key));
         }
 
-        let r = self
-            .region_ranges
-            .range((Excluded(enc_start_key(&snap_region)), Unbounded::<Key>))
-            .map(|(_, &region_id)| self.region_peers[&region_id].region())
-            .take_while(|r| enc_start_key(r) < enc_end_key(&snap_region))
-            .skip_while(|r| r.get_id() == region_id)
-            .next()
-            .map(|r| r.to_owned());
-        if let Some(exist_region) = r {
-            info!("region overlapped {:?}, {:?}", exist_region, snap_region);
-            let peer = &self.region_peers[&region_id];
-            // In some extreme case, it may happen that a new snapshot is received whereas a snapshot is still in applying
-            // if the snapshot under applying is generated before merge and the new snapshot is generated after merge,
-            // update `pending_cross_snap` here may cause source peer destroys itself improperly. So don't update
-            // `pending_cross_snap` here if peer is applying snapshot.
-            if !peer.is_applying_snapshot() && !peer.has_pending_snapshot() {
-                self.pending_cross_snap
-                    .insert(region_id, snap_region.get_region_epoch().to_owned());
-            }
-            self.raft_metrics.message_dropped.region_overlap += 1;
-            return Ok(Some(key));
-        }
         for region in &self.pending_snapshot_regions {
-            if enc_start_key(region) < enc_end_key(&snap_region) &&
-               enc_end_key(region) > enc_start_key(&snap_region) &&
+            if enc_start_key(region) < end_key &&
```

Comment:
Could you add a test case that can cover this line while without encode end key logic?

Reply:
It is covered by tests. If the end key is not in encoded form, the tests would fail:
    `failpoints_cases::test_merge::test_node_merge_multiple_snapshots_not_together`
    `failpoints_cases::test_merge::test_node_merge_multiple_snapshots_together`
    `failpoints_cases::test_snap::test_overlap_cleanup`